src/Juvix/FrontendContextualise/ModuleOpen/Transform.hs

```diff
@@ -53,92 +53,61 @@ restoreNameOpen ::
 restoreNameOpen (Just def, sym) = Env.addModMap sym def
 restoreNameOpen (Nothing, sym) = Env.removeModMap sym
 
+--------------------------------------------------
+-- Protection extension
+--------------------------------------------------
+protectOpen syms op =
+  protectSymbols syms $ do
+    originalQualified <- traverse saveOldOpen syms
+    traverse_ Env.removeModMap syms
+    res <- op
+    traverse restoreNameOpen originalQualified
+    pure res
 --------------------------------------------------
 -- Symbol Binding
 --------------------------------------------------
 
 transformLet :: Env.WorkingMaps m => Old.Let -> m New.Let
 transformLet (Old.LetGroup name bindings body) = do
-  originalVal <- Env.lookup name
-  qualifiedName <- Env.lookupModMap name
-  let transform = do
-        Env.addUnknown name
-        Env.removeModMap name
-        transformedBindings <- traverse transformFunctionLike bindings
-        let def = decideRecordOrDef transformedBindings Nothing
-        Env.add name def -- add to new context
-        New.LetGroup name transformedBindings <$> transformExpression body
-  res <- transform
-  restoreName (originalVal, name)
-  restoreNameOpen (qualifiedName, name)
-  pure res
+  protectOpen [name] $ do
+    transformedBindings <- traverse transformFunctionLike bindings
+    --
+    Env.add (NameSpace.Pub name) (decideRecordOrDef transformedBindings Nothing)
```

Comment:
Should this be considered public, it's a local add?

Reply:
Since protect just adds it to be unknown, we should add it to priv, since it gets precedence on lookup

Reply:
Added it to Priv, and removed it when the def is over